solidity/test/maintainer/MaintainerProxy.test.ts

```diff
@@ -0,0 +1,3434 @@
+                        await createSnapshot()
+
+                        outcome = runMovedFundsSweepScenario(data)
+                      })
+
+                      after(async () => {
+                        await restoreSnapshot()
+                      })
+
+                      it("should succeed", async () => {
+                        await expect(outcome).to.not.be.reverted
+                      })
+
+                      it("should refund ETH", async () => {
+                        const resolvedOutcome = await outcome
+
+                        const postThirdPartyBalance = await provider.getBalance(
+                          authorizedMaintainer.address
+                        )
+
+                        const diff = postThirdPartyBalance.sub(
+                          resolvedOutcome.initThirdPartyBalance
+                        )
+                        expect(diff).to.be.gt(0)
+                        expect(diff).to.be.lt(
+                          ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+                        )
+                      })
+                    }
+                  )
+                })
+
+                context("when the sweeping wallet has a main UTXO set", () => {
+                  context(
+                    "when the first input refers to a Pending sweep request and the second input refers to the sweeping wallet main UTXO",
+                    () => {
+                      const data: MovedFundsSweepTestData =
+                        MovedFundsSweepWithMainUtxo
+
+                      let outcome: Promise<ScenarioOutcome>
+
+                      before(async () => {
+                        await createSnapshot()
+
+                        outcome = runMovedFundsSweepScenario(data)
+                      })
+
+                      after(async () => {
+                        await restoreSnapshot()
+                      })
+
+                      it("should succeed", async () => {
+                        await expect(outcome).to.not.be.reverted
+                      })
+
+                      it("should refund ETH", async () => {
+                        const resolvedOutcome = await outcome
+
+                        const postThirdPartyBalance = await provider.getBalance(
+                          authorizedMaintainer.address
+                        )
+
+                        const diff = postThirdPartyBalance.sub(
+                          resolvedOutcome.initThirdPartyBalance
+                        )
+                        expect(diff).to.be.gt(0)
+                        expect(diff).to.be.lt(
+                          ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+                        )
+                      })
+                    }
+                  )
+                })
+              }
+            )
+          })
+        })
+
+        context("when sweeping wallet is in the MovingFunds state", () => {
```

Comment:
Let's drop this test, it's irrelevant for gas refunds. The processing happens exactly the same for Live and MovingFunds wallet.

Reply:
685430015da9d5bd9817c00b7eaf50c7f75e1b22